### PR TITLE
feat(sbb-form-field): reference label for custom form components

### PR DESCRIPTION
**Pull request by kyubisation**
Currently `<sbb-form-field>` only links native form elements correctly via the `<label>` `for` attribute. This PR adds the functionality for custom form elements to be able to reference the `<label>` via `aria-labelledby` on the custom form element, which is done internally in the `<sbb-form-field>`.

### CODE REVIEW

File: src/components/sbb-form-field/sbb-form-field.tsx [hunk 1]
```diff
@@ -196,10 +197,8 @@ export class SbbFormField implements ComponentInterface {
     if (labels.length > 1 && createdLabel) {
       createdLabel.remove();
     }
-    const inputId = this._input?.id;
-    if (inputId) {
-      labels.forEach((l) => (l.htmlFor = inputId));
-    }
+    this._labels = Array.from(this._element.querySelectorAll('label'));
```
Comment (jeripeierSBB) [inline src/components/sbb-form-field/sbb-form-field.tsx:200]: This assumes that label is slotted inside the component, but it can be anywhere in the light DOM?
```
<label for="select-1">Label</label>
<sbb-form-field>
  <sbb-select id="select-1"></sbb-select>
</sbb-form-field>
```

or more general (not form field related I know)
```
<label for="select-1">Label</label>
<sbb-select id="select-1"></sbb-select>
```

File: src/components/sbb-form-field/sbb-form-field.tsx [hunk 2]
```diff
@@ -225,15 +224,37 @@ export class SbbFormField implements ComponentInterface {
...
+  private _syncLabelInputReferences(): void {
+    if (!this._input || !this._labels.length) {
+      return;
     }
 
-    this._element.dataset.inputType = this._input.tagName.toLowerCase();
-    const label = this._element.querySelector('label');
-    if (label) {
-      label.htmlFor = this._input.id;
+    if (this._supportedNativeInputElements.includes(this._input.tagName)) {
```
Comment (jeripeierSBB) [inline src/components/sbb-form-field/sbb-form-field.tsx:236]: I think there would be some more branches and specialities which could be tested

File: src/components/sbb-datepicker/sbb-datepicker.e2e.ts [hunk 3]
```diff
@@ -129,7 +129,7 @@ describe('sbb-datepicker', () => {
 
   describe('with form-field', () => {
     const template = `
-      <sbb-form-field>
+      <sbb-form-field label="Example">
```
Comment (jeripeierSBB) [inline src/components/sbb-datepicker/sbb-datepicker.e2e.ts:132]: Why is this change needed?

File: src/components/sbb-form-field/sbb-form-field.tsx (hunk 1 above)
Comment (kyubisation) [reply src/components/sbb-form-field/sbb-form-field.tsx:200]: Needs documentation

File: src/components/sbb-form-field/sbb-form-field.tsx (hunk 2 above)
Comment (kyubisation) [reply src/components/sbb-form-field/sbb-form-field.tsx:236]: Refactor to only use one label and document accordingly

File: src/components/sbb-datepicker/sbb-datepicker.e2e.ts (hunk 3 above)
Comment (kyubisation) [reply src/components/sbb-datepicker/sbb-datepicker.e2e.ts:132]: Investigate again

File: src/components/sbb-form-field/readme.md [hunk 4]
```diff
@@ -71,8 +71,14 @@ element. However, several of the form field's optional features interact with th
 contained within the form field.
 
 When you provide a label via `<label>` or the `label` attribute, `<sbb-form-field>` automatically
-associates this label with the field's form element via a native <label> element, using the `for`
+associates this label with the field's form element via a native `<label>` element, using the `for`
 attribute to reference the control's ID.
+When using a non-native form element (e.g. `<sbb-select>`) `aria-label` is used, to connect the
```
Comment (jeripeierSBB) [inline src/components/sbb-form-field/readme.md:76]:
```suggestion
When using a non-native form element (e.g. `<sbb-select>`) `aria-labelledby` is used, to connect the
```

File: src/components/sbb-form-field/readme.md [hunk 5]
```diff
@@ -71,8 +71,14 @@ element. However, several of the form field's optional features interact with th
...
+form element with the label, by setting an id on the label and referencing this id in the
+`aria-label` attribute placed on the form element.
```
Comment (jeripeierSBB) [inline src/components/sbb-form-field/readme.md:78]:
```suggestion
`aria-labelledby` attribute placed on the form element.
```

**Review by jeripeierSBB — APPROVED** [review]
LGTM, not sure if 'aria-label' in documentation was on purpose.

File: src/components/sbb-form-field/readme.md (hunk 4 above)
Comment (kyubisation) [reply src/components/sbb-form-field/readme.md:76]: Nope, that's a mistake. Brain might have been afk. 😄
